test/utils/test_bit_register_mapping.py

```diff
@@ -0,0 +1,71 @@
+import math
+from collections.abc import Iterable
+
+import pytest
+
+from opensquirrel import CircuitBuilder
+from opensquirrel.default_instructions import measure
+from opensquirrel.ir import Measure
+from opensquirrel.utils.bit_register_mapping import BitStringMapping
+
+
+@pytest.fixture(name="bit_string_mapping")
+def bit_string_mapping_fixture() -> BitStringMapping:
+    return BitStringMapping()
+
+
+@pytest.mark.parametrize(
+    ("measurements", "expected_mapping", "expected_bit_allocation"),
+    [
+        ([measure(0, 0)], {0: [0]}, {0: 0}),
+        ([measure(0, 0), measure(1, 0)], {0: [0], 1: [0]}, {0: 1}),
+        ([measure(0, 1), measure(0, 0)], {0: [1, 0]}, {0: 0, 1: 0}),
+    ],
+)
+def test_add_measure(
+    bit_string_mapping: BitStringMapping,
+    measurements: Iterable[Measure],
+    expected_mapping: dict[int, list[int]],
+    expected_bit_allocation: dict[int, int],
```

Comment:
Can we define a type for bit allocations? Would it be useful outside tests?

Reply:
It is simply a mapping that maps classical bits to acquisition channels (qubit indices). In the output format each classical bit can only be used for one acquisition channel. This mapping is needed to see what we need should not export. It is only used for this purpose, so I'm not sure we need a type for it. It is only used in the tests and only makes sense with respect to this specific class.

Reply:
> It is simply a mapping that maps classical bits to acquisition channels (qubit indices).

Nobody knows that. Or has to know that. You won't know that in 3 months time. That's my whole point on using type aliases. If that line would read `expected_bit_allocation: MapBitQubit` or `map_bit_qubit_t` or whatever, it would be much more readable.